plaidml/edsl/edsl.h

```diff
@@ -1273,5 +1275,64 @@ inline Tensor pragma(const Tensor& tensor, const std::string& op, const PragmaAt
 ///
 inline Tensor trace(const Tensor& x, const std::string& msg) { return pragma(x, "trace", {{"msg", Value(msg)}}); }
 
+using LayerBodySingleFn = std::function<Tensor()>;
+using LayerBodyMultiFn = std::function<Tensors()>;
+
+inline Tensors layer(const std::string& op, const Tensors& operands, const Dictionary& attrs,
```

Comment:
Aren't the operands derivable via the use/def pairs that cross the block boundary (which we have to compute regardless)?  If so, why make the user explicitly list them?